ext/authx/Civi/Authx/Authenticator.php

```diff
@@ -259,7 +259,7 @@ protected function checkPolicy(AuthenticatorTarget $tgt) {
    */
   protected function login(AuthenticatorTarget $tgt) {
     $isSameValue = function($a, $b) {
-      return !empty($a) && (string) $a === (string) $b;
+      return (is_null($a) && is_null($b)) || (!empty($a) && (string) $a === (string) $b);
```

Comment:
The conditional expression as it was written before this PR was odd. It started by unnecessarily using `!empty()` on a variable that is always defined (so `!empty()` is a waste). Then it converts the values to strings in order to compare integers.
I don't know what all the possibilities are for `$a` and `$b`, but I feel like this would probably work fine, as it would correctly differentiate `null` from `0`, correctly match `null` with `null`, etc.:
```suggestion
      return ((string) $a === (string) $b);
```

Reply:
Certainly simpler!  Thanks

Reply:
> ... I don't know what all the possibilities are for $a and $b...

Yes, that's the rub. Historically, I think we've had cases where the meaning of `0` /`null` was somehow asymmetric. Maybe WP+D7 interpreted UID 0 differently? Or maybe UID `0` meant "anonymous" while CID `0` meant anonymous? But I feel like maybe there was some normalization patch at some point? It's a tricky area to be confident about.

Reply:
@totten in certain subsystems it is still asymmetric and actually by design. If you call `CRM_Core_Permission::get()` the $cid param defaults to `NULL` which means "current user" whereas `0` means "anonymous user".

Reply:
@totten IMO the default of `null` == "current user" is a handy feature for higher-level functions but once we get into lower levels it's best to resolve that default value so the value passed around is always an int >= 0.